app/pre_upgrade.go

```diff
@@ -78,6 +80,7 @@ func preUpgrade(c *cli.Context) error {
 	err = newPreUpgrader(namespace, lhClient, eventRecorder).Run()
 	if err != nil {
 		logrus.Warnf("Done with Run() ... err is %v", err)
+		time.Sleep(5 * time.Minute)
```

Comment:
@coderabbitai Any better way to flush the event?

Reply:
I have my doubts about this, because eventBroadcaster.Shutdown() is already called in the `defer`.  But @PhanLe1010 and I tried this in our sync-up call, and moving it inline makes no difference.  It appears to me that Shutdown() does not guarantee the synchronous flush of queued events.
To minimize the wait, I'll need to look over the interface to find an explicit flush, or else write a poll loop to wait until we can check that the event is there, along the lines suggested below.

Reply:
@james-munson 

Can we combine eventBroadcaster.Shutdown() and sleep like this?

```go
defer func() {
    eventBroadcaster.Shutdown()
    if err != nil {
        time.Sleep(5 * time.Second)
    }
}()
```

Reply:
That seems worth a try.  I'll give that a go.  The shorter sleep should be plenty.

Reply:
That seems to be sufficient.  It runs, events, and exits in a matter of seconds.  I'll push it up and see whether it is acceptable.